src/analysis/convex/cone.lean

```diff
@@ -634,13 +634,26 @@ lemma pointed_inner_dual_cone : s.inner_dual_cone.pointed :=
 lemma inner_dual_cone_eq_Inter_inner_dual_cone_singleton :
   (s.inner_dual_cone : set H) = ⋂ i : s, (({i} : set H).inner_dual_cone : set H) :=
 begin
-  simp_rw [set.Inter_coe_set, subtype.coe_mk],
-  refine set.ext (λ x, iff.intro (λ hx, _) _),
-  { refine set.mem_Inter.2 (λ i, set.mem_Inter.2 (λ hi _, _)),
-    rintro ⟨ ⟩,
-    exact hx i hi },
-  { simp only [set.mem_Inter, convex_cone.mem_coe, mem_inner_dual_cone,
-      set.mem_singleton_iff, forall_eq, imp_self] }
+  ext,
+  refine ⟨λ hx, set.mem_Inter_of_mem _, by simp⟩,
+  rintros i _ ⟨ ⟩,
+  exact hx i i.prop,
+end
+
+lemma closed_inner_dual_cone : is_closed (s.inner_dual_cone : set H) :=
+begin
+  -- reduce the problem to showing that dual cone of a singleton `{x}` is closed
+  rw inner_dual_cone_eq_Inter_inner_dual_cone_singleton,
+  apply is_closed_Inter,
+  intros x,
+
+  -- the dual cone of a singleton set is the preimage of `[0, ∞)` under `inner x`
+  have h : (({x} : set H).inner_dual_cone : set H) = (inner x : H → ℝ)⁻¹' (set.Ici 0) :=
+  by { ext, simp },
```

Comment:
This looks like a good standalone lemma.

Reply:
In fact, there's a more general version of this lemma that I added in #15836

Reply:
Thanks! I'll merge it here.

Reply:
You should now be able to `rw inner_dual_cone_singleton`.

Reply:
Updated.